File: 2023/javascript/day_01/day_1_2.js
const {readFileSync} = require("fs")

const input = readFileSync("./input.txt", "utf-8").replace(/\r/g, "")
const codes = input.split("\n")

const nums = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]

const numbers = []

for(let code of codes) {
    let first, last;

    for(let i = 0; i < code.length; i++) {
        let parsed = parseInt(code[i]);
        if(isNaN(parsed)) {
            const num = nums.find(n => code.substring(i).startsWith(n))
            parsed = nums.indexOf(num) + 1
        }

        if(parsed) {
            if(first === undefined) first = parsed
            last = parsed
        }
    }
    numbers.push(first * 10 + last)
}

const sum = numbers.reduce((a, b) => a + b)

console.log(`The sum of all numbers is: ${sum}`)

File: 2023/python/day_01/day_1_2.py
lines = open("input.txt", "r").read().replace(r"\r", "").split("\n")

numbers = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]

sum = 0


def get_num(s):
    if s[0].isnumeric():
        return int(s[0])

    i = 1
    res = None
    for n in numbers:
        if s.startswith(n):
            res = i
            break
        i += 1
    return res


for line in lines:
    first = None
    last = None
    for i in range(0, len(line)):
        num = get_num(line[i:len(line)])
        if num is None:
            continue
        if first is None:
            first = num
        last = num
    sum += (first * 10) + last

print(f"The sum of all calibration values is: {sum}")
